packages/aws-rfdk/lib/core/lambdas/nodejs/version-provider/version-provider.ts

```diff
@@ -0,0 +1,323 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable no-console */
+
+import * as fs from 'fs';
+import { IncomingMessage } from 'http';
+
+export enum Platform {
+  linux = 'linux',
+
+  mac = 'mac',
+
+  windows = 'windows',
+}
+
+export enum Product {
+  deadline = 'Deadline',
+
+  deadlineDocker = 'DeadlineDocker',
+}
+
+export interface IVersionProviderProperties {
+  readonly versionString?: string
+
+  readonly product: Product;
+
+  readonly platform?: Platform;
+}
+
+export interface IUris {
+}
+
+export interface IDeadlineInstallerUris extends IUris {
+  readonly bundle: string;
+
+  readonly clientInstaller: string;
+
+  readonly repositoryInstaller: string;
+
+  readonly certificateInstaller?: string;
```

Comment:
If we have to make the client and repository optional here to account for how our Mac installers work anyways, what do you think about getting rid of `IDeadlineDockerRecipeUris` and `IDeadlineInstallerUris` then moving these fields to `IUris`?

None of it is ideal, I still kind of think we should have a separate class for retrieving values from the DeadlineDocker section, but this would at least allow us to avoid having to make unsafe casts for the return types while leaving the logic in a single class.